dev/dfi/packet.c

```diff
@@ -89,47 +97,259 @@ wddr_return_t create_ck_packet_sequence(dfi_tx_packet_buffer_t *buffer,
+        addr_info->phase_remaining -= 1;
+        phase_offset++;
+    }
+#endif
+#if WDDR_PHY_MAX_FREQ_RATIO == WDDR_FREQ_RATIO_1TO4
+    if (mask & 0x4)
+    {
+        packet->packet.packet.cs_p4 = command->address[phase_offset].cs;
+        packet->packet.packet.cs_p5 = command->address[phase_offset].cs;
+        packet->packet.packet.address_p4 = command->address[phase_offset].ca_pins;
+        addr_info->phase_remaining -= 1;
+        phase_offset++;
+    }
+    if (mask & 0x8)
+    {
+        packet->packet.packet.cs_p6 = command->address[phase_offset].cs;
+        packet->packet.packet.cs_p7 = command->address[phase_offset].cs;
+        packet->packet.packet.address_p6 = command->address[phase_offset].ca_pins;
+        addr_info->phase_remaining -= 1;
+        phase_offset++;
+    }
+#endif
+}
+
+void fill_wrdata_en_packet(packet_item_t *packet,
+                           packet_group_info_t *wrdata_en_info,
+                           uint8_t cycles_per_packet)
+{
+    uint32_t mask = determine_group_mask(wrdata_en_info,
+                                         cycles_per_packet,
+                                         DATA_PHASE_PER_CYC_SHFT);
+
+    if (mask & 0x1)
+    {
+        packet->packet.packet.wrdata_en_p0 = 1;
+        packet->packet.packet.wrdata_en_p1 = 1;
+        wrdata_en_info->phase_remaining -= 2;
+    }
+#if WDDR_PHY_MAX_FREQ_RATIO >= WDDR_FREQ_RATIO_1TO2
+    if (mask & 0x2)
+    {
+        packet->packet.packet.wrdata_en_p2 = 1;
+        packet->packet.packet.wrdata_en_p3 = 1;
+        wrdata_en_info->phase_remaining -= 2;
+    }
+#endif
+#if WDDR_PHY_MAX_FREQ_RATIO == WDDR_FREQ_RATIO_1TO4
+    if (mask & 0x4)
+    {
+        packet->packet.packet.wrdata_en_p4 = 1;
+        packet->packet.packet.wrdata_en_p5 = 1;
+        wrdata_en_info->phase_remaining -= 2;
+    }
+    if (mask & 0x8)
+    {
+        packet->packet.packet.wrdata_en_p6 = 1;
+        packet->packet.packet.wrdata_en_p7 = 1;
+        wrdata_en_info->phase_remaining -= 2;
+    }
+#endif
+}
+
+void fill_wrdata_packet(packet_item_t *packet,
+                        packet_group_info_t *wrdata_info,
+                        chipselect_t cs,
+                        command_data_t *data,
+                        uint8_t cycles_per_packet,
+                        uint16_t phase_length)
+{
+    uint32_t mask = determine_group_mask(wrdata_info,
+                                         cycles_per_packet,
+                                         DATA_PHASE_PER_CYC_SHFT);
+
+    uint16_t data_offset = phase_length - wrdata_info->phase_remaining;
+
+    if (mask & 0x1)
+    {
+        packet->packet.packet.wrdata_cs_p0  = 1 << cs;
+        packet->packet.packet.dq0_wrdata_p0 = data->dq[0][data_offset];
+        packet->packet.packet.dq1_wrdata_p0 = data->dq[1][data_offset];
+        packet->packet.packet.dq0_wrdata_mask_p0 = data->dm[0][data_offset];
+        packet->packet.packet.dq1_wrdata_mask_p0 = data->dm[1][data_offset];
```

Comment:
Should we drop support for data mask? We're not actually using it. There might be on place where we create the data frame but I think it's not using it correctly anyway.

Reply:
So as of right now, I don't think we ever set the wrdata_mask values in the command_data_t structure before passing it in, so we can definitely remove it if not needed.  There is something nice about leaving it there just in case we need it in the future, but I'm having a hard time seeing when we will need it unless for testing functionality of the buffer.  I am not sure if the implementation is correct or not either right now in the way we would want to use it, but I think that would come more in the creation of the create data function filling in the command_data_t structure.

Reply:
Yeah, let's just remove it for now. I don't see us needing it for any of our algorithms and we're trying to only have minimal code required.

Reply:
Removed.